src/foam/core/Method.js

```diff
@@ -137,6 +138,7 @@ foam.CLASS({
 
     function installInClass(cls, superMethod, existingMethod) {
       var method = this;
+      method.parentCls = cls;
```

Comment:
We should do this in createChildMethod_ which should clone the axiom if it's going to mutate it.

installInClass/installInProto shouldn't mutate the axiom IMO.

Reply:
I agree with not wanting to mutate the axiom here but I need this applied to all methods, not just child methods. Do you know a better place for this?

Reply:
Reusing axioms can happen with implements: but implemens already clones its axioms to avoid them being shared. What is the parentCls needed for?

Reply:
It's needed for figuring out if a method overrides a parent method because swift needs the 'override' keyword and it's also used for providing a default implementation to methods that aren't implemented when a class implements an interface. Swift doesn't have abstract classes so we fake it by adding the methods that are missing with a 'fatalError()' implementation.

It's used in this file: https://github.com/foam-framework/foam2/blob/35dc0bd85e197cec26c06ad6c9da1d4d6ea8107b/src/foam/swift/refines/Method.js

I realize there are alternatives here, just not sure which route to take. Any suggestions?

Reply:
I removed the parentCls property from Method and exposed some methods instead:
https://github.com/foam-framework/foam2/pull/675/commits/0cf057b7826996d3a2a68fb3627a269d31e2180c

I'm not crazy about having to use helper methods for accessing these properties but I'm not sure if there's an alternative if I can't store this extra bit of state on the model. I was trying to think of an approach that uses the context instead but I'm not sure that's any better than my previous solution. Thoughts?